File: 271-encode-and-decode-strings/encode-and-decode-strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {

        string curr = "";
        for(string s1 : strs)
        {
            int len = s1.size();
            curr = curr + to_string(len) + "@" + s1;
        }
        return curr;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        
        int index = 0;
        vector<string>ans;
        while(index < s.size())
        {
            int pos = s.find_first_of('@', index);
            int len = stoi(s.substr(index, pos - index));
            index = pos + 1;
            ans.push_back(s.substr(index, len));
            index+=len;
        }
        return ans;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: 271-encode-and-decode-strings/encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
    
        encoded = ""
        for word in strs:
            encoded+=str(len(word)) + "@" + word
        return encoded
        

    def decode(self, s: str) -> List[str]:
        
        res = []
        i = 0
        while i < len(s):
            # find delimiter
            # the string before that is length of string
            # traverse length of string and add that to the answer
            index = s.find('@', i)
            #print(index)
            length = int(s[i:index])
            word = s[index+1 : index+1+length]
            #print(word)
            res.append(word)
            i = index+1+length
        return res


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))